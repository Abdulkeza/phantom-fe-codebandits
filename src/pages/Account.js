import React from 'react';
import { Link } from 'react-router-dom';
import SideBar from '../components/SideBar.js';

const Account = () => {
  return (
    <div className="min-h-[70vh]">
      <main className="flex flex-row h-full">
        <aside>
          <SideBar />
        </aside>
        <section className="w-7/10 bg-blue flex items-center justify-center  w-full flex-col">
          <h1 className="font-bold">
            This is the notification board for logged in user
          </h1>
          <div className="flex flex-col">
            <div className="flex">
              <Link
                to="/dashboard/driver/register"
                className="m-2 py-1 px-3 bg-primary rounded-md"
              >
                Register Driver
              </Link>
              <Link
                to="/dashboard/driver/update"
                className="ml-8 m-2 py-1 px-3 bg-gray-500 rounded-md"
              >
                Update Driver
              </Link>
            </div>
            <div className="flex">
              <div>
                <Link
                  to="/dashboard/operator/register"
                  className="m-2 py-1 px-3 bg-primary rounded-md"
                >
                  Register Operator
                </Link>
              </div>
              <Link
                to="/dashboard/operator/update"
                className="m-2 py-1 px-3 bg-gray-500 rounded-md"
              >
                Update Operator
              </Link>
            </div>
          </div>
        </section>
      </main>
    </div>
  );
};

export default Account;
