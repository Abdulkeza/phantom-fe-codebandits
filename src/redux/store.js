import { configureStore } from '@reduxjs/toolkit';
import { counterReducer } from './reducers/counterReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';
import { profileReducer } from './reducers/profileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer
  }
});
