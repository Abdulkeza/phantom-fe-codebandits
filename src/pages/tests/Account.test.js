import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../redux/store.js';
import Account from '../Account.js';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate
}));

describe('Account', () => {
  it('should render the account page', () => {
    const elem = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Account />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});

describe('Account functionality tests', () => {
  const logoutMock = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <Account logout={logoutMock()} />
      </MemoryRouter>
    </Provider>
  );
  const logoutBtn = wrapper.find('#logout');
  it('should call logout function when logout button clicked', () => {
    logoutBtn.simulate('click');
    expect(logoutMock).toHaveBeenCalled();
  });
});
