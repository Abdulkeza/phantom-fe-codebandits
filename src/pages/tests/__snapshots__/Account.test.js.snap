// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account should render the account page 1`] = `
<div>
  <div
    className="min-h-[70vh]"
  >
    <main
      className="flex flex-row h-full"
    >
      <aside>
        <div
          className="relative z-10"
        >
          <main
            className="hidden md:flex lg:flex xl:flex flex text-sm"
          >
            <aside
              className="border-r-2 h-full sticky top-0 px-8 py-4 bg-gray-100 transition-transform"
            >
              <div
                className=" mb-6"
              >
                <span
                  className="rounded-full mx-auto flex items-center justify-center h-12 w-12 mb-5"
                >
                  <img
                    alt="Image"
                    className="rounded-full object-cover h-12 w-12"
                    src="profileImg.jpg"
                  />
                </span>
                <div
                  className="text-center font-black text-md"
                >
                  <h1>
                    NGABO Patrick
                  </h1>
                  <div>
                    <h1
                      className="font-bold"
                    >
                      Operator
                    </h1>
                  </div>
                </div>
              </div>
              <div>
                <h2
                  className="mb-2 text-lg font-black"
                >
                  Account
                </h2>
                <ul
                  className="mb-2"
                >
                  <div
                    className="flex justify-center items-center hover:bg-gray-200 py-1"
                  >
                    <img
                      alt="icon"
                      className="w-6"
                      src="dashboardIcon.png"
                    />
                    <a
                      className="py-1 px-2 font-bold my-1  cursor-pointer w-full"
                      href="/account"
                      onClick={[Function]}
                    >
                      <li>
                        Dashboard
                      </li>
                    </a>
                  </div>
                  <div
                    className="flex justify-center items-center hover:bg-gray-200 py-1"
                  >
                    <img
                      alt="icon"
                      className="w-6"
                      src="manageIcon.png"
                    />
                    <a
                      aria-current="page"
                      className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
                      href="/#p"
                      onClick={[Function]}
                    >
                      <li>
                        Management
                      </li>
                    </a>
                  </div>
                  <div
                    className="flex justify-center items-center hover:bg-gray-200 py-1"
                  >
                    <img
                      alt="icon"
                      className="w-6"
                      src="profileIcon.png"
                    />
                    <a
                      className="py-1 px-2 font-bold my-1  cursor-pointer w-full"
                      href="/dashboard/profile"
                      onClick={[Function]}
                    >
                      <li>
                        Profile
                      </li>
                    </a>
                  </div>
                  <div
                    className="flex justify-center items-center hover:bg-gray-200 py-1"
                  >
                    <img
                      alt="icon"
                      className="w-6"
                      src="notificationIcon.png"
                    />
                    <a
                      aria-current="page"
                      className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
                      href="/#p"
                      onClick={[Function]}
                    >
                      <li>
                        Notifications
                      </li>
                    </a>
                  </div>
                  <div
                    className="flex justify-center items-center hover:bg-gray-200 py-1"
                  >
                    <img
                      alt="icon"
                      className="w-6"
                      src="supportIcon.png"
                    />
                    <a
                      aria-current="page"
                      className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
                      href="/#p"
                      onClick={[Function]}
                    >
                      <li>
                        Support
                      </li>
                    </a>
                  </div>
                  <div
                    className="flex justify-center items-center hover:bg-gray-200 py-1"
                  >
                    <img
                      alt="icon"
                      className="w-6"
                      src="settingIcon.png"
                    />
                    <a
                      aria-current="page"
                      className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
                      href="/#p"
                      onClick={[Function]}
                    >
                      <li>
                        Settings
                      </li>
                    </a>
                  </div>
                </ul>
                <div>
                  <button
                    className="bg-red px-6 py-1 hover:bg-red-600 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                    onClick={[Function]}
                    type="submit"
                  >
                    Logout
                  </button>
                </div>
              </div>
            </aside>
          </main>
          <div
            className="flex"
          >
            <button
              className="md:hidden lg:hidden xl:hidden"
              id="button"
              onClick={[Function]}
              type="button"
            >
              <div>
                <div
                  className="ml-2 bg-black h-1 w-7 bg-primary mt-1"
                />
                <div
                  className="ml-2 bg-black h-1 w-7 bg-primary mt-1"
                />
                <div
                  className="ml-2 bg-black h-1 w-7 bg-primary mt-1"
                />
              </div>
            </button>
          </div>
        </div>
      </aside>
      <section
        className="w-7/10 bg-blue flex items-center justify-center  w-full flex-col"
      >
        <h1
          className="font-bold"
        >
          This is the notification board for logged in user
        </h1>
        <div
          className="flex flex-col"
        >
          <div
            className="flex"
          >
            <a
              className="m-2 py-1 px-3 bg-primary rounded-md"
              href="/dashboard/driver/register"
              onClick={[Function]}
            >
              Register Driver
            </a>
            <a
              className="ml-8 m-2 py-1 px-3 bg-gray-500 rounded-md"
              href="/dashboard/driver/update"
              onClick={[Function]}
            >
              Update Driver
            </a>
          </div>
          <div
            className="flex"
          >
            <div>
              <a
                className="m-2 py-1 px-3 bg-primary rounded-md"
                href="/dashboard/operator/register"
                onClick={[Function]}
              >
                Register Operator
              </a>
            </div>
            <a
              className="m-2 py-1 px-3 bg-gray-500 rounded-md"
              href="/dashboard/operator/update"
              onClick={[Function]}
            >
              Update Operator
            </a>
          </div>
        </div>
      </section>
    </main>
  </div>
</div>
`;
